### DISCUSSION TITLE
How to set VLC and/or Media source local file ?

### DISCUSSION THREAD

**Discussion by michaeltoohig** [Palakis/obs-websocket#1030] [Q&A] [answered]
I'm using OBSv28.0.2 trying to add a media file to a media source/vlc source via websockets so I can automate playback of videos. I have successfully done this for OBSv24 with the old OBS-websockets server by setting a source settings with `{"local_file": "path/to/file.mp4"}` but I don't see this feature in the current documentation. 

I'm looking at the docs here:
https://github.com/obsproject/obs-websocket/blob/master/docs/generated/protocol.md#requests

But I do not see `SetSourceSettings` in the list. But in the issues from 2020 that was the request type recommended to do just that so it must have changed since then. That issue can be found here:
https://github.com/obsproject/obs-websocket/issues/125

So how can I send a request via websockets to set a playlist item in a Meida/VLC source and begin playing it?

---

So I added a generic events callback to see all the events. I added a media source then added a file and saw the following events.

```
New event! Type: InputCreated | Raw Data: {'defaultInputSettings': {'buffering_mb': 2, 'clear_on_media_end': True, 'is_local_file': True, 'linear_alpha': False, 'looping': False, 'reconnect_delay_sec': 10, 'restart_on_activate': True, 'speed_percent': 100}, 'inputKind': 'ffmpeg_source', 'inputName': 'vlc-feature', 'inputSettings': {}, 'unversionedInputKind': 'ffmpeg_source'}
New event! Type: SceneItemSelected | Raw Data: {'sceneItemId': 6, 'sceneName': 'feature'}
New event! Type: SceneItemCreated | Raw Data: {'sceneItemId': 6, 'sceneItemIndex': 0, 'sceneName': 'feature', 'sourceName': 'vlc-feature'}
New event! Type: MediaInputActionTriggered | Raw Data: {'inputName': 'vlc-feature', 'mediaAction': 'OBS_WEBSOCKET_MEDIA_INPUT_ACTION_RESTART'}
New event! Type: MediaInputPlaybackStarted | Raw Data: {'inputName': 'vlc-feature'}
```

But still not sure where to read about the `mediaAction` enum or how to set a file for playback with websockets. Did version 5.0 or what I am using built into OBSv28.0.2 have reduced functionality?

**Comment by michaeltoohig** [comment]
So from what I can gather so far. It appears the transition to v5 protocol is not quick and many previous features are not yet available so the solution is to use the 4.x compat in the mean time.

**Reply by tt2468** [reply]
That is not true. obs-websocket 5.0.0 was actually delayed somewhat due to our commitment to keeping feature parity with 4.x.

**Comment by tt2468** [comment, marked as answer]
The request you are looking for is called `SetInputSettings`.

In 5.x, we largely clarified what specific "source" requests do by using the terms `Input` (conventionally, misleadingly, called a `source` in normal OBS lingo), `Scene`, `Transition` and `Filter`. These are all "sources" in the underlying OBS API, even though they are largely very different.

Over the course of the release and hearing feedback, it's clear that people have a hard time understanding why things are called the way they are now. During initial planning stages I had tried to make it clear that common assumptions about 4.x should not be used on 5.x, but I think I did a bad job of continuing to convey that idea.

I've been wanting to spend some more time doing cleanup and improvements, specifically in the documentation sector. Unfortunately for that, my time largely depends on what I can spare from my other obligations, so things haven't progressed quite as fast as I've hoped on that front.

**Reply by michaeltoohig** [reply]
Thanks for clarifying. Ya, do find the new protocol docs a bit sparse in detail/examples and hard to find exactly what I was looking for. Compared to the 4.x protocol it appears to the uninformed as many features are not available in 5.x.

Digging into the google sheet that detailed the progress road map for 5.x lead me to believe the 5.x protocol was trying to reach feature parity with 4.x and I made my original response due to that misunderstanding.

**Reply by ChandlerHooley** [reply]
I know that you all are a long ways off of this... I am new to using the OBS-websocket module and am trying to use the obs-powershell version of it to adjust the file that is being played in an OBS media source. How would I go about using changing the settings to and setting a new file to be used?

Any help would be really appreciated!
